### DISCUSSION TITLE
What will be the maintenance phase for 1.x now that 2.x is released?

### DISCUSSION THREAD

**Discussion by mig5** [ory/hydra#3332] [Q&A] [answered]
Congratulations on the release of Hydra 2.0, team!

It's pretty common practice with such a major release (especially with so many breaking changes at API level etc) that an open source team or org might continue to maintain the previous major release line for a limited time. 

I won't be so bold as to ask for an LTS phase for 1.x :)

But I am hopeful that the Ory team would continue to backport any security patches to 1.x where it is necessary, for a limited time (say even 6 months? A year?)

This will help those of us with a bit of red tape risk-wise to plan for the upgrade (to give you a sense: I have 3 environments I have to get through before even getting it rolled out to production). Especially consider that it isn't just a Hydra upgrade, but also necessitates modifications to the Login&Consent app due to how the API's changed, as well as potentially other tools that have used the API to automate OAuth2.0 client registration (not dynamic registration but automated via other internal tools).

I've not seen anything mentioned, but it makes me nervous if 1.x is essentially abandonware in favour of 2.x *immediately*. 

Can we expect to have some limited support for 1.x as a transition phase?

Thanks for any reassurance and congrats again, I'm looking forward to trying 2.0 out.

**Comment by aeneasr** [comment, marked as answer]
Hi @mig5 :) To be honest, we do not plan to do any changes to 1.x any more. The most secure way to consume Ory technology is via the managed service we now provide at https://console.ory.sh. We do not have a financial incentive to keep v1.x maintained. Open source users typically do not fund the work we do. That's of course what open source implies and we are well aware that this is the path we chose for Ory :)

But that in turn influences our priorities heavily. In the end, we need to finance all the development we do in open source and all work that does not pay into that goal means that there is a decreasing likelyhood that we will be able to continue building more Ory open source.

Having said that, it will be a case-by-case decision. Ory has a very good security and open source track record and leadership. I think that we have tried to make right in terms of security engineering always. Most vulnerabilities (e.g. some nested dependency's regex DoS CVE, OpenSSL CVEs, ...) do not affect the actual service. If there truly is a significant vulnerability in Ory's codebase or code that Ory depends on and exposes, we will consider whether backporting that change to v1.x is worth the effort. But we can not guarantee it.

I hope this helps understand our position, while also giving you at least some confidence that we will try to do our best when such a problem surfaces!
